mutacc_auto/files/sbatch.py

```diff
@@ -25,59 +23,81 @@ def get_timestamp():
 MAIL_FAIL = 'FAIL'
 MAIL_END = 'END'
 
-#SOME CONSTAN SBATCH
-# (OPTION, VALUE, SEPARATOR)
+#SOME DEFAULT SBATCH OPTIONS
+# (OPTION, VALUE)
 HEADER_OPTIONS = (
-    ('-A', ACCOUNT, ' '),
-    ('-n', NODES, ' '),
-    ('-t', TIME, ' '),
-    ('-J', JOBNAME, ' '),
-    ('--qos', PRIORITY, '='),
-    ('--mail-type', MAIL_FAIL, '=' ),
-    ('--mail-type', MAIL_END, '=' )
+    ('A', ACCOUNT),
+    ('n', NODES),
+    ('t', TIME),
+    ('J', JOBNAME),
+    ('qos', PRIORITY),
+    ('mail-type', MAIL_FAIL),
+    ('mail-type', MAIL_END)
 )
 
+ACTIVATE_ENVIRONMENT = "source activate"
+
+STDERR_SUFFIX = "err"
+STDOUT_SUFFIX = "out"
+
+NEWLINE = "\n"
+
 class SbatchScript():
     """
         Context manager to write sbatch scripts
     """
 
     @staticmethod
-    def get_header(stdout_file, stderr_file, email):
+    def get_header(log_directory, email=None):
 
         """
             Returns header to be printed in sbatch scripts
 
             Args:
-                stdout_file (str): path to file where stdout is written
-                stderr_file (str): path to file where stderr is written
+                log_directory (path): path to dir for log files
                 email (str): email to notify if job fails, or completes
 
             Returns:
                 header (str): string with header, lines separated by newline
         """
 
+        #Instantiate header string
         header = ""
         #write all constant header options
         for option in HEADER_OPTIONS:
-            header += f"{HEADER_PREFIX} {option[0]}{option[2]}{option[1]}\n"
-        #write all variable header options
-        header += f"{HEADER_PREFIX} -e {stderr_file}\n"
-        header += f"{HEADER_PREFIX} -o {stdout_file}\n"
-        header += f"{HEADER_PREFIX} --mail-user={email}\n"
+            if len(option[0]) == 1:
+                header += f"{HEADER_PREFIX} -{option[0]} {option[1]}{NEWLINE}"
+            else:
+                header += f"{HEADER_PREFIX} --{option[0]}={option[1]}{NEWLINE}"
+
+        #make log files
+        log_directory = Path(log_directory)
+        if not log_directory.is_dir():
```

Comment:
Can't this check be handled by Click?

Reply:
Yes you are right! this is unnecessary code. will remove.